pkg/antctl/raw/upgrade/apistorage/command.go

```diff
@@ -0,0 +1,271 @@
+	if err != nil {
+		return nil, err
+	}
+	// If the user-provided name list of CRDs to upgrade is empty, upgrade all Antrea CRDs.
+	if crdNamesToUpgrade.Len() == 0 {
+		crdNamesToUpgrade = antreaCRDNames
+	} else {
+		// If the user-provided name list of CRDs to upgrade is not empty, and it contains CRDs without suffix
+		// "crd.antrea.io", skip these CRDs.
+		for name := range crdNamesToUpgrade.Difference(antreaCRDNames) {
+			fmt.Fprintf(writer, "Skip CRD %q which is not created by Antrea.\n", name)
+		}
+		// Only upgrade the CRDs with suffix "crd.antrea.io".
+		crdNamesToUpgrade = crdNamesToUpgrade.Intersection(antreaCRDNames)
+	}
+	return crdNamesToUpgrade, nil
+}
+
+// getCRDsToUpgrade gets a list of Antrea CRDs to upgrade.
+func getCRDsToUpgrade(writer io.Writer, k8sClient client.Client, crdNamesToUpgrade sets.Set[string]) ([]*apiextv1.CustomResourceDefinition, error) {
+	var crdsToUpgrade []*apiextv1.CustomResourceDefinition
+	for crdName := range crdNamesToUpgrade {
+		crd := &apiextv1.CustomResourceDefinition{}
+		if err := k8sClient.Get(context.TODO(), client.ObjectKey{Name: crdName}, crd); err != nil {
+			return nil, fmt.Errorf("error getting CRD %q: %w", crdName, err)
+		}
+		// Skip the CRD that has only one version.
+		if len(crd.Spec.Versions) == 1 {
+			fmt.Fprintf(writer, "Skip upgrading CRD %q since it only has one version.\n", crdName)
+			continue
+		}
+		// Skip the CRD that all stored objects are in the storage version.
+		if len(crd.Status.StoredVersions) == 1 && crd.Status.StoredVersions[0] == getCRDStorageVersion(crd) {
+			fmt.Fprintf(writer, "Skip upgrading CRD %q since all stored objects are in the storage version.\n", crdName)
+			continue
+		}
+		crdsToUpgrade = append(crdsToUpgrade, crd)
+	}
+	return crdsToUpgrade, nil
+}
+
+// upgradeCRDObjects upgrades the existing objects of a CRD.
+func upgradeCRDObjects(writer io.Writer, k8sClient client.Client, crd *apiextv1.CustomResourceDefinition) error {
+	objList := &unstructured.UnstructuredList{}
+	objList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   crd.Spec.Group,
+		Version: getCRDStorageVersion(crd),
+		Kind:    crd.Spec.Names.ListKind,
+	})
+
+	if err := k8sClient.List(context.TODO(), objList); err != nil {
+		return err
+	}
+
+	if len(objList.Items) == 0 {
+		return nil
+	}
+
+	fmt.Fprintf(writer, "Upgrading %d objects of CRD %q.\n", len(objList.Items), crd.Name)
+	for _, item := range objList.Items {
+		if err := upgradeCRDObject(writer, k8sClient, crd, item); err != nil {
+			if apierrors.IsNotFound(err) {
+				fmt.Fprintf(writer, "Skip upgrading object %s of CRD %q which is not found.\n", k8s.NamespacedName(item.GetNamespace(), item.GetName()), crd.Name)
+			} else {
+				return fmt.Errorf("error upgrading object %s of CRD %q: %w", k8s.NamespacedName(item.GetNamespace(), item.GetName()), crd.Name, err)
+			}
+		}
+	}
+	fmt.Fprintf(writer, "Successfully upgraded %d objects of CRD %q.\n", len(objList.Items), crd.Name)
```

Comment:
You call `len(objList.Items)` multiple times,  please define a variable to reuse.